infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -141,275 +134,317 @@ private boolean isBootstrapped(CorfuMsg msg, ChannelHandlerContext ctx, IServerR
     /**
      * Handle a layout request message.
      *
-     * @param msg              corfu message containing LAYOUT_REQUEST
+     * @param req              corfu message containing LAYOUT_REQUEST
      * @param ctx              netty ChannelHandlerContext
      * @param r                server router
      */
-    @ServerHandler(type = CorfuMsgType.LAYOUT_REQUEST)
-    public synchronized void handleMessageLayoutRequest(CorfuPayloadMsg<Long> msg,
-                                                    ChannelHandlerContext ctx, IServerRouter r) {
-        if (!isBootstrapped(msg, ctx, r)) {
+    @RequestHandler(type = RequestPayloadMsg.PayloadCase.LAYOUT_REQUEST)
+    public synchronized void handleLayoutRequest(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
```

Comment:
I know this predates your patch, but honestly this doesn't make sense at all: having a multi-threaded executor to server the request handlers and marking all the handlers as synchronized is just lol. 

Can you just change `this.executor`  to be single threaded and drop all the synchronized ?

Reply:
With Biased locking having synchronized without contention should be cheap, so you can keep the synchronized methods, but change the executor to a single thread.

I would just remove the synchronized though, it seems super unnecessary.

Reply:
Removed the synchronized, thanks!